src/core/Dropdown/Dropdown/Dropdown.tsx

```diff
@@ -241,37 +177,60 @@ export class Dropdown extends Component<DropdownProps> {
         id={id}
         {...passProps}
       >
-        <HtmlLabel
-          id={labelId}
-          {...labelProps}
-          className={dropdownClassNames.label}
-        >
-          {labelMode === 'hidden' ? (
-            <VisuallyHidden>{labelText}</VisuallyHidden>
-          ) : (
-            <Paragraph {...labelTextProps}>{labelText}</Paragraph>
-          )}
-        </HtmlLabel>
-        <ListboxInput {...listboxInputProps}>
-          <ListboxButton {...passDropdownButtonProps}>
-            {listboxDisplayValue}
-          </ListboxButton>
-          {!!ListboxPopoverComponentReplace ? (
-            <ListboxPopoverComponentReplace {...passDropdownPopoverProps}>
-              {this.dropdownItems(children, passDropdownItemProps)}
-            </ListboxPopoverComponentReplace>
-          ) : (
+        <HtmlDiv>
+          <LabelText
+            id={labelId}
+            labelMode={labelMode}
+            as="label"
+            optionalText={optionalText}
+          >
+            {labelText}
+          </LabelText>
+          <ListboxInput {...listboxInputProps}>
```

Comment:
Could these props be told right here I the template for ListBoxInput, as this is the only place they are used?

https://github.com/vrk-kpa/suomifi-ui-components/blob/91b2a1fbc8770b5771e3a16cc211f1a412d90802/src/core/Dropdown/Dropdown/Dropdown.tsx#L158-L164

Reply:
Yes they could. And now they are :)